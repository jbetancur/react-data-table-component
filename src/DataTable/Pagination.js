import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { useTableContext } from './DataTableContext';
import Select from './Select';
import { getNumberOfPages, detectRTL } from './util';
import useWindowSize from '../hooks/useWindowSize';
import { media, SMALL } from './media';

const defaultComponentOptions = {
  rowsPerPageText: 'Rows per page:',
  rangeSeparatorText: 'of',
  noRowsPerPage: false,
  selectAllRowsItem: false,
  selectAllRowsItemText: 'All',
};

const PaginationWrapper = styled.nav`
  display: flex;
  flex: 1 1 auto;
  justify-content: flex-end;
  align-items: center;
  box-sizing: border-box;
  padding-right: 8px;
  padding-left: 8px;
  width: 100%;
  ${props => props.theme.pagination.style};
`;

const Button = styled.button`
  position: relative;
  display: block;
  user-select: none;
  border: none;
  ${props => props.theme.pagination.pageButtonsStyle};
  ${props => props.isRTL && 'transform: scale(-1, -1)'};
`;

const PageList = styled.div`
  display: flex;
  align-items: center;
  border-radius: 4px;
  white-space: nowrap;
  ${media.sm`
    width: 100%;
    justify-content: space-around;
  `};
`;

const Span = styled.span`
  flex-shrink: 1;
  user-select: none;
`;

const Range = styled(Span)`
  margin: 0 24px;
`;

const RowLabel = styled(Span)`
  margin: 0 4px;
`;

const Pagination = ({
  rowsPerPage,
  rowCount,
  onChangePage,
  onChangeRowsPerPage,
  currentPage,
}) => {
  const {
    data,
    direction,
    paginationRowsPerPageOptions,
    paginationIconLastPage,
    paginationIconFirstPage,
    paginationIconNext,
    paginationIconPrevious,
    paginationComponentOptions,
  } = useTableContext();
  const windowSize = useWindowSize();
  const shouldShow = windowSize.width > SMALL;
  const isRTL = detectRTL(direction);
  const numPages = getNumberOfPages(rowCount, rowsPerPage);
  const lastIndex = currentPage * rowsPerPage;
  const firstIndex = (lastIndex - rowsPerPage) + 1;
  const disabledLesser = currentPage === 1;
  const disabledGreater = currentPage === numPages;
  const options = { ...defaultComponentOptions, ...paginationComponentOptions };
  const range = currentPage === numPages
    ? `${firstIndex}-${rowCount} ${options.rangeSeparatorText} ${rowCount}`
    : `${firstIndex}-${lastIndex} ${options.rangeSeparatorText} ${rowCount}`;

  const handlePrevious = useCallback(() => onChangePage(currentPage - 1), [currentPage, onChangePage]);
  const handleNext = useCallback(() => onChangePage(currentPage + 1), [currentPage, onChangePage]);
  const handleFirst = useCallback(() => onChangePage(1), [onChangePage]);
  const handleLast = useCallback(() => onChangePage(getNumberOfPages(rowCount, rowsPerPage)), [onChangePage, rowCount, rowsPerPage]);
  const handleRowsPerPage = useCallback(({ target }) => onChangeRowsPerPage(Number(target.value), currentPage), [currentPage, onChangeRowsPerPage]);

  const selectOptions = paginationRowsPerPageOptions.map(num => (
    <option
      key={num}
      value={num}
    >
      {num}
    </option>
  ));

  if (options.selectAllRowsItem) {
    selectOptions.push(
      (
        <option
          key={-1}
          value={data.length}
        >
          {options.selectAllRowsItemText}
        </option>
      ),
    );
  }

  const select = (
    <Select onChange={handleRowsPerPage} defaultValue={rowsPerPage}>
      {selectOptions}
    </Select>
  );

  return (
    <PaginationWrapper className="rdt_Pagination">
      {!options.noRowsPerPage && shouldShow && (
        <>
          <RowLabel>{options.rowsPerPageText}</RowLabel>
          {select}
        </>
      )}
      {shouldShow && (
        <Range>
          {range}
        </Range>
      )}
      <PageList>
        <Button
          id="pagination-first-page"
          type="button"
          aria-label="First Page"
          aria-disabled={disabledLesser}
          onClick={handleFirst}
          disabled={disabledLesser}
          isRTL={isRTL}
        >
          {paginationIconFirstPage}
        </Button>

        <Button
          id="pagination-previous-page"
          type="button"
          aria-label="Previous Page"
          aria-disabled={disabledLesser}
          onClick={handlePrevious}
          disabled={disabledLesser}
          isRTL={isRTL}
        >
          {paginationIconPrevious}
        </Button>

        {!shouldShow && select}

        <Button
          id="pagination-next-page"
          type="button"
          aria-label="Next Page"
          aria-disabled={disabledGreater}
          onClick={handleNext}
          disabled={disabledGreater}
          isRTL={isRTL}
        >
          {paginationIconNext}
        </Button>

        <Button
          id="pagination-last-page"
          type="button"
          aria-label="Last Page"
          aria-disabled={disabledGreater}
          onClick={handleLast}
          disabled={disabledGreater}
          isRTL={isRTL}
        >
          {paginationIconLastPage}
        </Button>
      </PageList>
    </PaginationWrapper>
  );
};

Pagination.propTypes = {
  rowsPerPage: PropTypes.number.isRequired,
  rowCount: PropTypes.number.isRequired,
  onChangePage: PropTypes.func.isRequired,
  onChangeRowsPerPage: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired,
};

export default Pagination;
