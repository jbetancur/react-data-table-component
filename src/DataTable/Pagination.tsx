import * as React from 'react';
import styled from 'styled-components';
import Select from './Select';
import { getNumberOfPages } from './util';
import useWindowSize from '../hooks/useWindowSize';
import useRTL from '../hooks/useRTL';
import { media, SMALL } from './media';
import { Direction } from './constants';
import { PaginationOptions } from './types';
import { defaultProps } from './defaultProps';

const defaultComponentOptions = {
	rowsPerPageText: 'Rows per page:',
	rangeSeparatorText: 'of',
	noRowsPerPage: false,
	selectAllRowsItem: false,
	selectAllRowsItemText: 'All',
};

const PaginationWrapper = styled.nav`
	display: flex;
	flex: 1 1 auto;
	justify-content: flex-end;
	align-items: center;
	box-sizing: border-box;
	padding-right: 8px;
	padding-left: 8px;
	width: 100%;
	${({ theme }) => theme.pagination.style};
`;

const Button = styled.button<{
	isRTL: boolean;
}>`
	position: relative;
	display: block;
	user-select: none;
	border: none;
	${({ theme }) => theme.pagination.pageButtonsStyle};
	${({ isRTL }) => isRTL && 'transform: scale(-1, -1)'};
`;

const PageList = styled.div`
	display: flex;
	align-items: center;
	border-radius: 4px;
	white-space: nowrap;
	${media.sm`
    width: 100%;
    justify-content: space-around;
  `};
`;

const Span = styled.span`
	flex-shrink: 1;
	user-select: none;
`;

const Range = styled(Span)`
	margin: 0 24px;
`;

const RowLabel = styled(Span)`
	margin: 0 4px;
`;

interface PaginationProps {
	rowsPerPage: number;
	rowCount: number;
	currentPage: number;
	direction?: Direction;
	paginationRowsPerPageOptions?: number[];
	paginationIconLastPage?: React.ReactNode;
	paginationIconFirstPage?: React.ReactNode;
	paginationIconNext?: React.ReactNode;
	paginationIconPrevious?: React.ReactNode;
	paginationComponentOptions?: PaginationOptions;
	onChangePage: (page: number) => void;
	onChangeRowsPerPage: (numRows: number, currentPage: number) => void;
}

function Pagination({
	rowsPerPage,
	rowCount,
	currentPage,
	direction = defaultProps.direction,
	paginationRowsPerPageOptions = defaultProps.paginationRowsPerPageOptions,
	paginationIconLastPage = defaultProps.paginationIconLastPage,
	paginationIconFirstPage = defaultProps.paginationIconFirstPage,
	paginationIconNext = defaultProps.paginationIconNext,
	paginationIconPrevious = defaultProps.paginationIconPrevious,
	paginationComponentOptions = defaultProps.paginationComponentOptions,
	onChangeRowsPerPage = defaultProps.onChangeRowsPerPage,
	onChangePage = defaultProps.onChangePage,
}: PaginationProps): JSX.Element {
	const windowSize = useWindowSize();
	const isRTL = useRTL(direction);
	const shouldShow = windowSize.width && windowSize.width > SMALL;
	// const isRTL = detectRTL(direction);
	const numPages = getNumberOfPages(rowCount, rowsPerPage);
	const lastIndex = currentPage * rowsPerPage;
	const firstIndex = lastIndex - rowsPerPage + 1;
	const disabledLesser = currentPage === 1;
	const disabledGreater = currentPage === numPages;
	const options = { ...defaultComponentOptions, ...paginationComponentOptions };
	const range =
		currentPage === numPages
			? `${firstIndex}-${rowCount} ${options.rangeSeparatorText} ${rowCount}`
			: `${firstIndex}-${lastIndex} ${options.rangeSeparatorText} ${rowCount}`;

	const handlePrevious = React.useCallback(() => onChangePage(currentPage - 1), [currentPage, onChangePage]);
	const handleNext = React.useCallback(() => onChangePage(currentPage + 1), [currentPage, onChangePage]);
	const handleFirst = React.useCallback(() => onChangePage(1), [onChangePage]);
	const handleLast = React.useCallback(
		() => onChangePage(getNumberOfPages(rowCount, rowsPerPage)),
		[onChangePage, rowCount, rowsPerPage],
	);
	const handleRowsPerPage = React.useCallback(
		(e: React.ChangeEvent<HTMLSelectElement>) => onChangeRowsPerPage(Number(e.target.value), currentPage),
		[currentPage, onChangeRowsPerPage],
	);

	const selectOptions = paginationRowsPerPageOptions.map((num: number) => (
		<option key={num} value={num}>
			{num}
		</option>
	));

	if (options.selectAllRowsItem) {
		selectOptions.push(
			<option key={-1} value={rowCount}>
				{options.selectAllRowsItemText}
			</option>,
		);
	}

	const select = (
		<Select onChange={handleRowsPerPage} defaultValue={rowsPerPage} aria-label={options.rowsPerPageText}>
			{selectOptions}
		</Select>
	);

	return (
		<PaginationWrapper className="rdt_Pagination">
			{!options.noRowsPerPage && shouldShow && (
				<>
					<RowLabel>{options.rowsPerPageText}</RowLabel>
					{select}
				</>
			)}
			{shouldShow && <Range>{range}</Range>}
			<PageList>
				<Button
					id="pagination-first-page"
					type="button"
					aria-label="First Page"
					aria-disabled={disabledLesser}
					onClick={handleFirst}
					disabled={disabledLesser}
					isRTL={isRTL}
				>
					{paginationIconFirstPage}
				</Button>

				<Button
					id="pagination-previous-page"
					type="button"
					aria-label="Previous Page"
					aria-disabled={disabledLesser}
					onClick={handlePrevious}
					disabled={disabledLesser}
					isRTL={isRTL}
				>
					{paginationIconPrevious}
				</Button>

				{!shouldShow && select}

				<Button
					id="pagination-next-page"
					type="button"
					aria-label="Next Page"
					aria-disabled={disabledGreater}
					onClick={handleNext}
					disabled={disabledGreater}
					isRTL={isRTL}
				>
					{paginationIconNext}
				</Button>

				<Button
					id="pagination-last-page"
					type="button"
					aria-label="Last Page"
					aria-disabled={disabledGreater}
					onClick={handleLast}
					disabled={disabledGreater}
					isRTL={isRTL}
				>
					{paginationIconLastPage}
				</Button>
			</PageList>
		</PaginationWrapper>
	);
}

export default React.memo(Pagination);
