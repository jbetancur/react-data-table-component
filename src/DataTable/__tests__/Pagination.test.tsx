/**
 * @jest-environment jsdom
 */

import 'jest-styled-components';
import * as React from 'react';
import { fireEvent } from '@testing-library/react';
import { renderWithTheme } from '../../internal/test-helpers'; // since child elements require theme
import Pagination from '../Pagination';

test('should render correctly with default props', () => {
	const { container } = renderWithTheme(
		<Pagination
			currentPage={1}
			rowsPerPage={10}
			rowCount={40}
			onChangePage={jest.fn()}
			onChangeRowsPerPage={jest.fn()}
		/>,
	);

	expect(container.firstChild).toMatchSnapshot();
});

describe('when clicking the First Page button', () => {
	test('should call onChangePage with correct args when currentPage is > 1', () => {
		const onChangePageMock = jest.fn();
		const { container } = renderWithTheme(
			<Pagination
				currentPage={4}
				rowsPerPage={10}
				rowCount={40}
				onChangePage={onChangePageMock}
				onChangeRowsPerPage={jest.fn()}
			/>,
		);

		fireEvent.click(container.querySelector('button#pagination-first-page') as HTMLButtonElement);
		expect(onChangePageMock).toBeCalledWith(1);
	});

	test('should NOT call onChangePage with correct args when currentPage is === 1', () => {
		const onChangePageMock = jest.fn();
		const { container } = renderWithTheme(
			<Pagination
				currentPage={1}
				rowsPerPage={10}
				rowCount={40}
				onChangePage={onChangePageMock}
				onChangeRowsPerPage={jest.fn()}
			/>,
		);

		fireEvent.click(container.querySelector('button#pagination-first-page') as HTMLButtonElement);
		expect(onChangePageMock).not.toBeCalled();
	});
});

describe('when clicking the Last Page button', () => {
	test('should call onChangePage with correct args when currentPage is < the last page', () => {
		const onChangePageMock = jest.fn();
		const { container } = renderWithTheme(
			<Pagination
				currentPage={1}
				rowsPerPage={10}
				rowCount={40}
				onChangePage={onChangePageMock}
				onChangeRowsPerPage={jest.fn()}
			/>,
		);

		fireEvent.click(container.querySelector('button#pagination-last-page') as HTMLButtonElement);
		expect(onChangePageMock).toBeCalledWith(4);
	});

	test('should NOT call onChangePage with correct args when currentPage is the last page', () => {
		const onChangePageMock = jest.fn();
		const { container } = renderWithTheme(
			<Pagination
				currentPage={4}
				rowsPerPage={10}
				rowCount={40}
				onChangePage={onChangePageMock}
				onChangeRowsPerPage={jest.fn()}
			/>,
		);

		fireEvent.click(container.querySelector('button#pagination-last-page') as HTMLButtonElement);
		expect(onChangePageMock).not.toBeCalled();
	});
});

describe('when clicking the Next Page button', () => {
	test('should call onChangePage with correct args when currentPage is not the last page', () => {
		const onChangePageMock = jest.fn();
		const { container } = renderWithTheme(
			<Pagination
				currentPage={1}
				rowsPerPage={10}
				rowCount={40}
				onChangePage={onChangePageMock}
				onChangeRowsPerPage={jest.fn()}
			/>,
		);

		fireEvent.click(container.querySelector('button#pagination-next-page') as HTMLButtonElement);
		expect(onChangePageMock).toBeCalledWith(2);
	});

	test('should NOT call onChangePage with correct args when currentPage is the last page', () => {
		const onChangePageMock = jest.fn();
		const { container } = renderWithTheme(
			<Pagination
				currentPage={4}
				rowsPerPage={10}
				rowCount={40}
				onChangePage={onChangePageMock}
				onChangeRowsPerPage={jest.fn()}
			/>,
		);

		fireEvent.click(container.querySelector('button#pagination-next-page') as HTMLButtonElement);
		expect(onChangePageMock).not.toBeCalled();
	});
});

describe('when clicking the Previous Page button', () => {
	test('should call onChangePage with correct args when currentPage is > 1', () => {
		const onChangePageMock = jest.fn();
		const { container } = renderWithTheme(
			<Pagination
				currentPage={2}
				rowsPerPage={10}
				rowCount={40}
				onChangePage={onChangePageMock}
				onChangeRowsPerPage={jest.fn()}
			/>,
		);

		fireEvent.click(container.querySelector('button#pagination-previous-page') as HTMLButtonElement);
		expect(onChangePageMock).toBeCalledWith(1);
	});

	test('should NOT call onChangePage with correct args when currentPage is 1', () => {
		const onChangePageMock = jest.fn();
		const { container } = renderWithTheme(
			<Pagination
				currentPage={1}
				rowsPerPage={10}
				rowCount={40}
				onChangePage={onChangePageMock}
				onChangeRowsPerPage={jest.fn()}
			/>,
		);

		fireEvent.click(container.querySelector('button#pagination-previous-page') as HTMLButtonElement);
		expect(onChangePageMock).not.toBeCalled();
	});
});

describe('when there is no paging to be done', () => {
	test('should NOT call onChangePage with correct with any nav action when there are less rows that the page size', () => {
		const onChangePageMock = jest.fn();
		const { container } = renderWithTheme(
			<Pagination
				currentPage={1}
				rowsPerPage={10}
				rowCount={5}
				onChangePage={onChangePageMock}
				onChangeRowsPerPage={jest.fn()}
			/>,
		);

		fireEvent.click(container.querySelector('button#pagination-last-page') as HTMLButtonElement);
		expect(onChangePageMock).not.toBeCalled();

		onChangePageMock.mockReset();

		fireEvent.click(container.querySelector('button#pagination-first-page') as HTMLButtonElement);
		expect(onChangePageMock).not.toBeCalled();

		onChangePageMock.mockReset();

		fireEvent.click(container.querySelector('button#pagination-previous-page') as HTMLButtonElement);
		expect(onChangePageMock).not.toBeCalled();

		onChangePageMock.mockReset();

		fireEvent.click(container.querySelector('button#pagination-next-page') as HTMLButtonElement);
		expect(onChangePageMock).not.toBeCalled();
	});
});

describe('when clicking the Previous Page button', () => {
	test('should call onChangePage with correct args when currentPage is > 1', () => {
		const onChangeRowsPerPageMock = jest.fn();
		// TOOO: remove when trailing empty rows is fixed
		const onChangePageMock = jest.fn();
		const { container } = renderWithTheme(
			<Pagination
				currentPage={2}
				rowsPerPage={10}
				rowCount={40}
				onChangePage={onChangePageMock}
				onChangeRowsPerPage={onChangeRowsPerPageMock}
			/>,
		);

		fireEvent.change(container.querySelector('select') as HTMLSelectElement, { target: { value: 20 } });
		expect(onChangeRowsPerPageMock).toBeCalledWith(20, 2);
	});
});
