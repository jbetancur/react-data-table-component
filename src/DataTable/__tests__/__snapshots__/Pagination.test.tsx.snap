// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with default props 1`] = `
.c4 {
  cursor: pointer;
  height: 24px;
  max-width: 100%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding-left: 8px;
  padding-right: 24px;
  box-sizing: content-box;
  font-size: inherit;
  color: inherit;
  border: none;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  direction: ltr;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4::-ms-expand {
  display: none;
}

.c4:disabled::-ms-expand {
  background: #f60;
}

.c4 option {
  color: initial;
}

.c3 {
  position: relative;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: inherit;
  color: inherit;
  margin-top: 1px;
}

.c3 svg {
  top: 0;
  right: 0;
  color: inherit;
  position: absolute;
  fill: currentColor;
  width: 24px;
  height: 24px;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  padding-right: 8px;
  padding-left: 8px;
  width: 100%;
  color: rgba(0,0,0,0.54);
  font-size: 13px;
  min-height: 56px;
  background-color: #FFFFFF;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: rgba(0,0,0,.12);
}

.c7 {
  position: relative;
  display: block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  border-radius: 50%;
  height: 40px;
  width: 40px;
  padding: 8px;
  margin: px;
  cursor: pointer;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  color: rgba(0,0,0,.54);
  fill: rgba(0,0,0,.54);
  background-color: transparent;
}

.c7:disabled {
  cursor: unset;
  color: rgba(0,0,0,.18);
  fill: rgba(0,0,0,.18);
}

.c7:hover:not(:disabled) {
  background-color: rgba(0,0,0,.12);
}

.c7:focus {
  outline: none;
  background-color: rgba(0,0,0,.12);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  white-space: nowrap;
}

.c1 {
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5 {
  margin: 0 24px;
}

.c2 {
  margin: 0 4px;
}

@media screen and (max-width:599px) {
  .c6 {
    width: 100%;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

<nav
  class="c0 rdt_Pagination"
>
  <span
    class="c1 c2"
  >
    Rows per page:
  </span>
  <div
    class="c3"
  >
    <select
      aria-label="Rows per page:"
      class="c4"
    >
      <option
        selected=""
        value="10"
      >
        10
      </option>
      <option
        value="15"
      >
        15
      </option>
      <option
        value="20"
      >
        20
      </option>
      <option
        value="25"
      >
        25
      </option>
      <option
        value="30"
      >
        30
      </option>
    </select>
    <svg
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
  <span
    class="c1 c5"
  >
    1-10 of 40
  </span>
  <div
    class="c6"
  >
    <button
      aria-disabled="true"
      aria-label="First Page"
      class="c7"
      disabled=""
      id="pagination-first-page"
      type="button"
    >
      <svg
        aria-hidden="true"
        height="24"
        role="presentation"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
        />
        <path
          d="M24 24H0V0h24v24z"
          fill="none"
        />
      </svg>
    </button>
    <button
      aria-disabled="true"
      aria-label="Previous Page"
      class="c7"
      disabled=""
      id="pagination-previous-page"
      type="button"
    >
      <svg
        aria-hidden="true"
        height="24"
        role="presentation"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
    <button
      aria-disabled="false"
      aria-label="Next Page"
      class="c7"
      id="pagination-next-page"
      type="button"
    >
      <svg
        aria-hidden="true"
        height="24"
        role="presentation"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
    <button
      aria-disabled="false"
      aria-label="Last Page"
      class="c7"
      id="pagination-last-page"
      type="button"
    >
      <svg
        aria-hidden="true"
        height="24"
        role="presentation"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
    </button>
  </div>
</nav>
`;
